packages/@lwc/engine-core/src/framework/utils.ts

```diff
@@ -101,23 +99,6 @@ export function flattenStylesheets(stylesheets: TemplateStylesheetFactories): St
     return list;
 }
 
-// Set a ref (lwc:ref) on a VM, from a template API
-export function setRefVNode(vm: VM, ref: string, vnode: VBaseElement | VStatic) {
-    if (process.env.NODE_ENV !== 'production' && isUndefined(vm.refVNodes)) {
-        throw new Error('refVNodes must be defined when setting a ref');
-    }
-
-    // If this method is called, then vm.refVNodes is set as the template has refs.
-    // If not, then something went wrong and we threw an error above.
-    const refVNodes: RefVNodes = vm.refVNodes!;
-
-    // In cases of conflict (two elements with the same ref), prefer, the last one,
-    // in depth-first traversal order.
-    if (!(ref in refVNodes) || refVNodes[ref].key < vnode.key) {
-        refVNodes[ref] = vnode;
-    }
```

Comment:
I probably need to confirm that, by removing this check and just using the normal rendering order, I didn't introduce an observable change into how conflicting `lwc:ref`s are resolved.

It turns out that this logic was wrong anyway, because we cannot assume that `key`s are in depth-first traversal order. Sometimes the key is a string: https://github.com/salesforce/lwc/blob/9e3bdd4cead3060fed3da662d43f73e67cb3a426/packages/%40lwc/template-compiler/src/codegen/index.ts#L628-L635

This might not be super-observable anyway, because 1) refs are a new, under-utilized features, and 2) it's pretty uncommon for them to conflict in the first place. But it would be nice to have a consistent ordering here and to know that it hasn't changed.

Reply:
OK actually, I guess this isn't an issue, because those string keys only apply to _default slotted content_, and [we don't allow `lwc:ref`s on that](https://stackblitz.com/edit/salesforce-lwc-5meatb?file=src%2Fmodules%2Fx%2Fslottable%2Fslottable.html,src%2Fmodules%2Fx%2Fslottable%2Fslottable.js&title=LWC%20playground). I.e. this will throw a compile-time error:

```html
<template>
  <slot name="bar" lwc:ref="foo"></slot>  
</template>
```

...which is good, because that `<slot>` has a string key, so the ordering would be based on the slot name (`bar`) and not the tree-traversal order.